server/src/library/libraryUtils.ts

```diff
@@ -70,7 +92,24 @@ function getOrCreateLibraryDefinitions(
         definitionsByLocale.set(locale, mapped);
     }
 
-    return PQP.MapUtils.assertGet(definitionsByLocale, locale);
+    let result: Map<string, PQLS.Library.TLibraryDefinition> = PQP.MapUtils.assertGet(definitionsByLocale, locale);
+
+    // we got module library for this definition call
+    if (currentModuleLibraryJson.length) {
```

Comment:
How often will this condition be true? If that's a regular occurrence I don't love how we're shallow copying the map each call.

Reply:
true, we replaced this with a shallow copy of its parent, plus there is also another level of cache happening at `vscode-powerquery/server/src/server.tsL#393`, which could also shrink the times we hit with this shallow copy